### PR TITLE
[MNT-23821] Filter custom aspects

### PR DESCRIPTION
**Pull request by tiagosalvado10**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [X] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

Only standard aspects are filtered based on `aspect-visible` whitelist configuration, whereas the custom aspects are always shown.

**What is the new behaviour?**

Standard and custom aspects are now filtered based on `aspect-visible` whitelist configuration.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by DenysVuika** [comment]
@tiagosalvado10 unit tests are missing for this PR

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        const visibleAspects = ["std:standardAspectOne"];
+        spyOn(aspectsApi, 'listAspects').and.returnValue(Promise.resolve(standardAspectPagingMock));
+        aspectListService.getStandardAspects(visibleAspects).subscribe((response) => {
+            expect(response).toEqual([stdAspect1]);
+            done();
+        });
+    });
+
+    it('should get two standard aspects', (done) => {
+        const visibleAspects = ["std:standardAspectTwo", "std:standardAspectThree"];
```
Comment (DenysVuika) [inline lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts:53]: should be single quotes as per js/ts code styles

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        spyOn(aspectsApi, 'listAspects').and.returnValue(Promise.resolve(standardAspectPagingMock));
+        aspectListService.getStandardAspects(visibleAspects).subscribe((response) => {
+            expect(response).toEqual([stdAspect2, stdAspect3]);
+            done();
+        });
+    });
+
+    it('should get one custom aspect', (done) => {
+        const visibleAspects = ["cst:customAspectTwo"];
```
Comment (DenysVuika) [inline lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts:62]: should be single quotes as per js/ts code styles

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        spyOn(aspectsApi, 'listAspects').and.returnValue(Promise.resolve(customAspectPagingMock));
+        aspectListService.getCustomAspects(visibleAspects).subscribe((response) => {
+            expect(response).toEqual([cstAspect2]);
+            done();
+        });
+    });
+
+    it('should get two custom aspects', (done) => {
+        const visibleAspects = ["cst:customAspectOne", "cst:customAspectThree"];
```
Comment (DenysVuika) [inline lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts:71]: should be single quotes as per js/ts code styles

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        });
+
+        aspectListService = TestBed.inject(AspectListService);
+        apiService = TestBed.inject(AlfrescoApiService);
+        aspectsApi = new AspectsApi(apiService.getInstance());
+        spyOnProperty(aspectListService, 'aspectsApi', 'get').and.returnValue(aspectsApi);
+    });
+
+    it('should get one standard aspect', (done) => {
+        const visibleAspects = ["std:standardAspectOne"];
```
[the remark this answers is not in the document]
Comment (tiagosalvado10) [reply lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts]: Applied change from double to single quotes.

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts (hunk 1 above)
Comment (tiagosalvado10) [reply lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts]: Applied change from double to single quotes.

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts (hunk 2 above)
Comment (tiagosalvado10) [reply lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts]: Applied change from double to single quotes.

File: lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts (hunk 3 above)
Comment (tiagosalvado10) [reply lib/content-services/src/lib/aspect-list/services/aspect-list.service.spec.ts]: Applied change from double to single quotes.

**Review by DenysVuika — APPROVED** [review]

**Comment by DenysVuika** [comment]
needs to be rebased with latest develop

**Pull request merged**